### PR TITLE
Update Storage to azure-core preview 3 

### PR DESCRIPTION
**Pull request by jianghaolu**

### CODE REVIEW

**Comment by jianghaolu** [comment]
Closing in favor of https://github.com/Azure/azure-sdk-for-java/pull/4489

File: storage/client/file/swagger/file.json [hunk 1]
```diff
@@ -3443,6 +3620,15 @@
...
             }
         },
+        "FilePath": {
+            "name": "filePath",
+            "in": "path",
+            "description": "The path of the target file.",
+            "required": true,
+            "type": "string",
+            "x-ms-parameter-location": "method",
+            "x-ms-skip-url-encoding": true
```
Comment (jianghaolu) [inline storage/client/file/swagger/file.json]: James will discuss with team

File: storage/client/file/swagger/file.json [hunk 2]
```diff
@@ -3219,8 +3382,10 @@
                     "format": "etag"
                 },
                 "Quota": {
-                    "type": "integer",
-                    "format": "int64",
+                    "type": "integer"
```
Comment (jianghaolu) [inline storage/client/file/swagger/file.json]: James to discuss with team

File: storage/client/blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java [hunk 3]
```diff
@@ -98,7 +98,7 @@
     public BlockBlobAsyncClient asBlockBlobAsyncClient() {
         return new BlockBlobAsyncClient(new AzureBlobStorageBuilder()
             .url(getBlobUrl().toString())
-            .pipeline(azureBlobStorage.httpPipeline()), snapshot);
+            .pipeline(azureBlobStorage.getHttpPipeline()), snapshot);
```
Comment (jianghaolu) [inline storage/client/blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java]: Jianghao to start discussion with James and Jonathan

File: eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml [hunk 4]
```diff
@@ -445,16 +445,6 @@
     <Bug pattern="NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE"/>
   </Match>
 
-  <!-- Incorrect flagging, if the response is null a HttpResponseException should be thrown
-  https://github.com/azure/azure-sdk-for-java/issues/4556 -->
-  <Match>
-    <Or>
-      <Class name="com.azure.security.keyvault.keys.KeyClient"/>
```
Comment (g2vinay) [inline eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml]: how was this spotbug issue resolved?

Comment (jianghaolu) [reply eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml]: This was brought in from merge - will revert.

**Comment by kurtzeborn** [comment]
We discussed this in our storage sync today and this can be merged now with follow up issues opened for the two follow ups:
1. Figure out something for the URL encoding in the files client
2. File quotas is in bytes so we need to measure with int64s instead of int32s

**Pull request merged**